include/boost/geometry/algorithms/detail/calculate_sum.hpp

```diff
@@ -30,7 +34,7 @@ class calculate_polygon_sum
     template <typename ReturnType, typename Policy, typename Rings, typename Strategy>
     static inline ReturnType sum_interior_rings(Rings const& rings, Strategy const& strategy)
     {
-        ReturnType sum = ReturnType();
+        ReturnType sum = ReturnType(0);
```

Comment:
Do you have some example where this is a problem?

For scalar types there is no problem. But for classes with user defined default constructor this constructor will be called. And it is possible that the type will not be initialized properly or will be initialized to some unexpected value like NaN in the constructor. Like in the case of the original ttmath type. This is why you had to implement some workarounds for this library in the past. I think that if an object has to be initialized explicitly to store some specific value then it should be explicitly stated in the code. So it's ok as it is or it could be even simpler:

    ReturnType sum = 0;

It'll compile if there is a valid chain of conversions to ReturnType. If not, then we cannot rely on the default constructor because anything may happen inside.